rollup/rollup_sync_service/rollup_sync_service.go

```diff
@@ -225,22 +225,33 @@ func (s *RollupSyncService) parseAndUpdateRollupEventLogs(logs []types.Log, endB
 			batchIndex := event.BatchIndex.Uint64()
 			log.Trace("found new FinalizeBatch event", "batch index", batchIndex)
 
-			parentBatchMeta, chunks, err := s.getLocalInfoForBatch(batchIndex)
-			if err != nil {
-				return fmt.Errorf("failed to get local node info, batch index: %v, err: %w", batchIndex, err)
+			lastFinalizedBatchIndex := rawdb.ReadLastFinalizedBatchIndex(s.db)
+			if lastFinalizedBatchIndex == nil {
```

Comment:
for smooth migration? can add a comment here to illustrate the expected case that `lastFinalizedBatchIndex` is `nil`.

Reply:
comments added and fixed a bug here. 
when lastFinalizedBatchIndex is nil, set it to current batchIndex minus 1

Reply:
got it.

Reply:
oh got it. then you also need another fix: handle batch index 0.

Reply:
I guess batch index 0 is automatically set in genesis and need not to be finalized right?

Reply:
that's true, but it will also emit a `FinalizeBatch` event. in current rollup-sync-service's implementation, it does not differentiate these two cases.

Reply:
It also applies to the case when batch index is 0
https://go.dev/play/p/l6gmkedw5S2

Reply:
discussed with @amoylan2, when batch index is 0, first `0 - 1` then `0 + 1` by two overflows will get 0, so resolve this thread.